src/main/kotlin/me/alexflipnote/kawaiibot/utils/BlackjackSession.kt

```diff
@@ -0,0 +1,138 @@
+package me.alexflipnote.kawaiibot.utils
+
+import com.google.common.collect.Sets
+import me.aurieh.ichigo.core.CommandContext
+import me.aurieh.ichigo.extensions.await
+import kotlin.math.min
+
+class BlackjackSession(private val ctx: CommandContext) {
+    private val ranks = setOf("A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K")
+    private val suits = setOf("♣", "♦", "♥", "♠")
+
+    private var playing = true
+    private var endStatus = "playing"
+
+    private var deck = mutableListOf<String>()
+
+    private val userCards = mutableListOf<String>()
+    private val botCards = mutableListOf<String>()
+
+    suspend fun start() {
+        initialCards()
+        while (playing) {
+            ctx.channel.sendMessage(getLayoutText())
+            if (getPoints(userCards) != 21) {
+                askCard()
+            }
+
+            val msg = ctx.channel.sendMessage("Do you want to play again? Type `yes` if you do.").await()
+            val choice = ctx.waitFor(ctx.sameContext(), 60).message.contentRaw
+            if (choice != "yes") {
+                playing = false
+            } else {
+                deck.clear()  // Remove this line for actual blackjack instead of pure randomness (requested by alex)
```

Comment:
@AlexFlipnote any comment on this?

Reply:
I mentioned that I didn't want a predictable blackjack card deck, as the original was 100% random.